ToxicR workinprogress.R

```diff
@@ -0,0 +1,51 @@
+library("devtools")
+#requires dependency Cmake
+#devtools::install_github("NIEHS/ToxicR")
+library("ToxicR")
+library("dplyr")
+library("PMCMRplus")
+library(parallel)
+library("pbmcapply")
+#bmdoutput <- read.table("~/storage/biomarker_input_.txt",
+#                        header=FALSE, sep = "\t", row.names = 1)
+bmdoutput <- read.table("~/storage/ToxicR mod/data for BMDExpress log2 CPM Males.txt",
+                        header=FALSE, sep = "\t", row.names = 1)
+#
+#bmdoutput <- bmdoutput[-3,-c(1:3,5:9,11:15,17:18)]
+#
+bmdoutput <- as.data.frame(t(bmdoutput))
+bmdoutput <-bmdoutput[,-1]
+bmdoutput[-1] <- lapply(bmdoutput[-1], function(x) {
+  as.numeric(as.character(x))
+})
+
+dose_column <-as.numeric(bmdoutput[,1]) 
+start.time <- Sys.time()
+Func <- function(resp_bygene_column){ 
+  trendtestres<- williamsTest(resp_bygene_column ~ dose_column, alternative = c("greater", "less"))
+  trendestresdf <- data_frame("pass/fail value" = trendtestres$crit.value[,1] - trendtestres$statistic[,1]) 
+  #only generate model when significant based on williams test 
+  if (any(trendestresdf$`pass/fail value` <0)){ 
+    model<-single_continuous_fit(dose_column,resp_bygene_column)
```

Comment:
It doesn't look like you're specifying any model; I guess the default is Hill?

The ideal way to approach this would be to wrap the whole function so that it can iterate over model types (ideally also in parallel)

Reply:
Yes the default was hill, I have implemented modelling of all 5 available models, and to save the best one based on The maximum value of the likelihod/posterior, I read that this is an assessment of the goodness of fit and a higher maximum would be considered the best model, please correct me if I am wrong. I have also implemented this in parallel processing (thank you for teaching me about it, it has changed my world), I used nested mclapplys but was worried about how it would be affected given the cores distributed to the workers (maybe there is an optimal split?), I also used foreach dopar, based on this discussion: https://stackoverflow.com/questions/51707443/should-mclapply-calls-be-nested. Indeed after testing, foreach dopar is slightly faster than nested mclapply with maximum cores used for both.

Reply:
output file also now contains the gene the bmd and the model that was the best used